lib/ansible/modules/cloud/amazon/cloudfront_invalidation.py

```diff
@@ -0,0 +1,275 @@
+                                 "To make a new invalidation please update caller_reference.")
+                return current_invalidation_response, False
+            else:
+                self.module.fail_json_aws(e, msg="Error creating CloudFront invalidations.")
+
+    def get_invalidation(self, distribution_id, caller_reference):
+        current_invalidation = {}
+        # find all invalidations for the distribution
+        try:
+            paginator = self.client.get_paginator('list_invalidations')
+            invalidations = paginator.paginate(DistributionId=distribution_id).build_full_result()['InvalidationList'].get('Items', [])
+            invalidation_ids = [inv['Id'] for inv in invalidations]
+        except (BotoCoreError, ClientError) as e:
+            self.module.fail_json_aws(e, msg="Error listing CloudFront invalidations.")
+
+        # check if there is an invalidation with the same caller reference
+        for inv_id in invalidation_ids:
+            try:
+                invalidation = self.client.get_invalidation(DistributionId=distribution_id, Id=inv_id)['Invalidation']
+                caller_ref = invalidation.get('InvalidationBatch', {}).get('CallerReference')
+            except (BotoCoreError, ClientError) as e:
+                self.module.fail_json_aws(e, msg="Error getting Cloudfront invalidation {0}".format(inv_id))
+            if caller_ref == caller_reference:
+                current_invalidation = invalidation
+                break
+
+        current_invalidation.pop('ResponseMetadata', None)
+        return current_invalidation
+
+
+class CloudFrontInvalidationValidationManager(object):
+    """
+    Manages Cloudfront validations for invalidation batches
+    """
+
+    def __init__(self, module):
+        self.module = module
+        self.__cloudfront_facts_mgr = CloudFrontFactsServiceManager(module)
+
+    def validate_distribution_id(self, distribution_id, alias):
+        try:
+            if distribution_id is None and alias is None:
+                self.module.fail_json(msg="distribution_id or alias must be specified")
+            if distribution_id is None:
+                distribution_id = self.__cloudfront_facts_mgr.get_distribution_id_from_domain_name(alias)
+            return distribution_id
+        except (ClientError, BotoCoreError) as e:
+            self.module.fail_json_aws(e, msg="Error validating parameters.")
+
+    def create_aws_list(self, invalidation_batch):
+        aws_list = {}
+        aws_list["Quantity"] = len(invalidation_batch)
+        aws_list["Items"] = invalidation_batch
+        return aws_list
+
+    def validate_invalidation_batch(self, invalidation_batch, caller_reference):
+        try:
+            if caller_reference is not None:
+                valid_caller_reference = caller_reference
+            else:
+                valid_caller_reference = datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f')
```

Comment:
For future reference, `datetime.datetime.now().isoformat()` will do the same thing as this `strftime` usage.

Reply:
Thanks for the tip.